app/models/concerns/set_perishable_token.rb

```diff
@@ -0,0 +1,14 @@
+module SetPerishableToken
```

Comment:
Maybe just nitpicking; I'm not sure if ```SetPerishableToken``` is a good name for a module. Maybe just ```PerishableToken``` instead?